client/python/openlineage/client/transport/msk_iam.py

```diff
@@ -0,0 +1,113 @@
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import functools
+import logging
+import os
+from typing import Any
+
+import attr
+from openlineage.client.transport import KafkaConfig, KafkaTransport
+
+log = logging.getLogger(__name__)
+
+
+def _detect_running_region() -> None | str:
+    """Dynamically determine the region from a running Glue job (or anything on EC2 for
+    that matter).
+    https://stackoverflow.com/questions/37514810/how-to-get-the-region-of-the-current-user-from-boto
+    """
+    import boto3  # type: ignore[import]
+
+    easy_checks = [
+        # check if set through ENV vars
+        os.environ.get("AWS_REGION"),
+        os.environ.get("AWS_DEFAULT_REGION"),
+        # else check if set in config or in boto already
+        boto3.DEFAULT_SESSION.region_name if boto3.DEFAULT_SESSION else None,
+        boto3.Session().region_name,
+    ]
+    region: str
+    for region in easy_checks:
+        if region:
+            return region
+
+    return None
+
+
+@attr.s
+class MSKIAMConfig(KafkaConfig):
+    # MSK producer config
+    # https://github.com/aws/aws-msk-iam-sasl-signer-python
+
+    region: str = attr.ib(default=None)
+    aws_profile: None | str = attr.ib(default=None)
+    role_arn: None | str = attr.ib(default=None)
+    aws_debug_creds: bool = attr.ib(default=False)
+
+
+def _oauth_cb(config: MSKIAMConfig, *_: Any) -> tuple[str, float]:
+    from aws_msk_iam_sasl_signer import MSKAuthTokenProvider  # type: ignore[import]
+
+    region = config.region
+    if config.aws_profile:
+        auth_token, expiry_ms = MSKAuthTokenProvider.generate_auth_token_from_profile(
+            region, config.aws_profile
+        )
+    elif config.role_arn:
+        auth_token, expiry_ms = MSKAuthTokenProvider.generate_auth_token_from_role_arn(
+            region, config.role_arn
+        )
+    # Implement the version to load a custom `botocore.credentials.CredentialProvider` at runtime
+    # and calling the method
+    # `MSKAuthTokenProvider.generate_auth_token_from_credentials_provider(region, credentials_provider)`
+    else:
+        auth_token, expiry_ms = MSKAuthTokenProvider.generate_auth_token(
+            region, aws_debug_creds=config.aws_debug_creds
+        )
+    log.debug("Token expiry time: %s region %s", expiry_ms, region)
+    # Note that this library expects oauth_cb to return expiry time in seconds since epoch,
+    # while the token generator returns expiry in ms
+    return auth_token, expiry_ms / 1000
+
+
+class MSKIAMTransport(KafkaTransport):
+    kind = "msk-iam"
+    config_class = MSKIAMConfig
+
+    def __init__(self, config: MSKIAMConfig) -> None:
+        self.msk_config = config
+        super().__init__(config)
+
+    def _setup_producer(self, config: dict) -> None:  # type: ignore[type-arg]
+        try:
+            log.info("Setup the MSK transport with this configuration: %s", self.msk_config)
```

Comment:
I think all other transport stick to the debug level, even when logging some configuration details. Should we stick to that?

Reply:
ok I didn't notice, I will do it.